lib/ecto/migration.ex

```diff
@@ -613,7 +613,20 @@ defmodule Ecto.Migration do
     add(column, :naive_datetime, opts)
   end
 
-  def add(column, type, opts) when is_atom(column) do
+  def add(column, type, opts) when is_atom(column) and type in [:decimal, :numeric] do
```

Comment:
I don't think we need to do it only for those types, it can be for any type. :) Before `validate_type!(type)` you could do this:

```elixir
if opts[:scale] && !opts[:precision] do
  raise ...
end
```